### PR TITLE
Fix typing and ``postinit`` in ``build_class``

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] Write a good description on what the PR does.

## Description

Follow up to #1433.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Related Issue

### CODE REVIEW

File: astroid/raw_building.py [hunk 1]
```diff
@@ -116,20 +116,22 @@ def build_module(name: str, doc: Optional[str] = None) -> nodes.Module:
     return node
 
 
-def build_class(name, basenames=(), doc=None):
-    """create and initialize an astroid ClassDef node"""
+def build_class(
+    name: str, basenames: Sequence[str] = (), doc: Optional[str] = None
```
Comment (cdce8p) [inline astroid/raw_building.py:120]:
```suggestion
    name: str, basenames: Iterable[str] = (), doc: Optional[str] = None
```
If you start that way, we should use the least limiting base class. 😉

File: astroid/raw_building.py [hunk 2]
```diff
@@ -116,20 +116,22 @@ def build_module(name: str, doc: Optional[str] = None) -> nodes.Module:
...
+    """Create and initialize an astroid ClassDef node."""
     node = nodes.ClassDef(name)
+    basenodes: List[nodes.Name] = []
+    for base in basenames:
+        basenode = nodes.Name(name=base)
+        basenode.parent = node
+        basenodes.append(basenode)
     node.postinit(
-        bases=[],
+        bases=basenodes,
```
Comment (cdce8p) [inline astroid/raw_building.py:130]:
```suggestion
        bases=[nodes.Name(name=base, parent=node) for base in basenames],
```

File: astroid/raw_building.py [hunk 3]
```diff
@@ -116,20 +116,22 @@ def build_module(name: str, doc: Optional[str] = None) -> nodes.Module:
...
+def build_class(
+    name: str, basenames: Sequence[str] = (), doc: Optional[str] = None
+) -> nodes.ClassDef:
+    """Create and initialize an astroid ClassDef node."""
     node = nodes.ClassDef(name)
+    basenodes: List[nodes.Name] = []
+    for base in basenames:
+        basenode = nodes.Name(name=base)
+        basenode.parent = node
+        basenodes.append(basenode)
```
Comment (cdce8p) [inline astroid/raw_building.py:128]:
```suggestion
```

File: astroid/raw_building.py [hunk 4]
```diff
@@ -32,7 +32,7 @@
 import sys
 import types
 import warnings
-from typing import List, Optional
+from typing import List, Optional, Sequence
```
Comment (cdce8p) [inline astroid/raw_building.py:35]:
```suggestion
from typing import Iterable, List, Optional
```

**Review by cdce8p — CHANGES_REQUESTED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Review by cdce8p — APPROVED** [review]

**Pull request merged**